easybuild/framework/easyconfig/style.py

```diff
@@ -0,0 +1,113 @@
+##
+# Copyright 2016 Ghent University
+#
+# This file is part of EasyBuild,
+# originally created by the HPC team of Ghent University (http://ugent.be/hpc/en),
+# with support of Ghent University (http://ugent.be/hpc),
+# the Flemish Supercomputer Centre (VSC) (https://vscentrum.be/nl/en),
+# the Hercules foundation (http://www.herculesstichting.be/in_English)
+# and the Department of Economy, Science and Innovation (EWI) (http://www.ewi-vlaanderen.be/en).
+#
+# http://github.com/hpcugent/easybuild
+#
+# EasyBuild is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation v2.
+#
+# EasyBuild is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with EasyBuild.  If not, see <http://www.gnu.org/licenses/>.
+##
+"""
+Style tests for easyconfig files using pep8.
+
+@author: Ward Poelmans (Ghent University)
+"""
+
+import re
+from vsc.utils import fancylogger
+
+from easybuild.tools.utilities import only_if_module_is_available
+
+try:
+    import pep8
+except ImportError:
+    pass
+
+_log = fancylogger.getLogger('easyconfig.style', fname=False)
+
+
+# any function starting with _eb_check_ will be added to the tests
+# if the test number is added to the select list. The test number is
+# definied as AXXX where A is either E or W and XXX is a 3 digit number.
+# It should be mentioned in the docstring as a single word.
+# Read the pep8 docs to understand the arguments of these functions
+def _eb_check_trailing_whitespace(physical_line, lines, line_number, total_lines):
+    """
+    W299
+    Warn about trailing whitespace, expect for the description and comments.
+    This differs from the standard trailing whitespace check as that
+    will warn for any trailing whitespace.
+    """
+    comment_re = re.compile('^\s*#')
+    if comment_re.match(physical_line):
+        return None
```

Comment:
not a big fan of inline returns (we rarely do this in the current codebase, I'd prefer sticking to that)